docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+
+Service catalog controllers vary, however, from ingress controllers, because a cluster can consist of many different controllers, each of which handles actions on a subset of the available service-related resources. For example, a cluster can run a service controller to handle AWS RDS databases, and a completely separate one to handle AWS Elastic Load Balancer instances.
+
+As we'll detail below, each service controller implementation must to conform to an implementation spec, but that spec allows for a heterogenous set of controllers in a cluster. We believe this allowance encourages flexibility in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how a `StorageClass` defines a _kind_ of persistent volume, but we expect that, in practice, the kinds of services represented here will be more coarse grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a service controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceInstanceClaim`
+  - Created by a consumer to bind to an existing ServiceInstance by name. As written above, this workflow enables multi-tenant usage of a resource
+  - Created by a consumer to provision and bind to an existing ServiceClass. As written above, this workflow enables dedicated, single-tenant usage of a resource
+- `ServiceInstanceBinding`
+  - Created in response to a successful bind operation
+  - References credentials created as a result of the bind operation
+  - Holds a record of an application that’s bound to a service
+
+
+The below diagram shows the interactions between these resources:
+
+![Resource Interactions](./img/partial-flow.png)
+
+# Example Resources
+
+## ServiceClass
+
+```yaml
+kind: ServiceClass
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: postgres-small
+spec:
+  serviceID: c5a0ad8e-b57b…
+  planID: c68cd6b1-be9f…
+```
+
+## ServiceInstance
+
+```yaml
+kind: ServiceInstance
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: zip-code-db
+spec:
+  serviceClass: postgres-small
+```
+
```

Comment:
@judkowitz all of your bullet points reflect this doc, however the nomenclature is a bit misrepresentative. The second bullet point says "...every time I do a claim on a type I need to go through the create instance process" - that should instead say "every time I do a claim on a _service class_ I need to go through the _provision_ process.

Further comments below:

>It's conflating the instantiation process with the single vs. multi-tenancy nature of the service

I believe this conflating is necessary.

The CF broker API defines an API for instantiation already, which dictates the following applicable operations: `provision`, `bind`, `unbind` and `deprovision`. These operations imply the tenancy of a resource.

We've modeled this design after those operations such that a `ServiceInstance` creation matches a `provision` and `ServiceInstanceClaim` may match a provision or provision+bind, depending on what it references. To repeat what you've said above, if the claim references a `ServiceClass`, it represents a `provision`+`bind`. If it references a `ServiceInstance`, it represents only a `bind`.

While I can think of some alternatives that separate instantiation with tenancy, I believe they make core operations more difficult. I'm willing to be convinced otherwise though!

>what if there is an operator and multiple developers? The operator will want to instantiate the service on behalf of developers with 0 bindings at the time of instantiation. That operator may intend for the instance to be just for one developer or for many.

In the first example, the operator would create a `ServiceInstance` manually. That triggers the `provision` operation. The `ServiceInstance` then has 0 apps bound to it. Later, developers can submit claims against that `ServiceInstance`, each of which triggers a `bind` operation. 

In the second example, developers may submit claims directly against the `ServiceClass`, which ensures that the instance is shared by nobody.

n.b.: There has been discussion in this PR around splitting claims into 2 different resource types: `ServiceInstanceClaim`, which is used for multi-tenancy only, and `ServiceClassClaim`, which is used for single-tenancy (we also call this "dedicated") only